thrust/thrust/system/detail/generic/swap_ranges.inl

```diff
@@ -48,9 +48,8 @@ struct swap_pair_elements
   template <typename Tuple>
   _CCCL_HOST_DEVICE void operator()(Tuple t)
   {
-    // use unqualified swap to allow ADL to catch any user-defined swap
-    using thrust::swap;
-    swap(thrust::get<0>(t), thrust::get<1>(t));
+    // fully qualify call to swap for two cuda::std::tuples to avoid ambiguity with thrust::swap
+    ::cuda::std::swap(thrust::get<0>(t), thrust::get<1>(t));
```

Comment:
I am having difficulties getting this correct without changes in Thrust as well, since here we want to swap any type from either namespace `thrust` or `cuda::std`.

The old version of the code now creates an ambiguity when swapping `tuple_of_iterator_references`, because both of the generic `thrust::swap` and `cuda::std::swap` match equally and for some reason the tiebreaker does not work.

In the new version, we fail to pick up user provided `swap`s using ADL.

Reply:
Yeah combining the two PRs seems reasonable